File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.py
n = int(input())
dasom = int(input())
others = sorted((int(input()) for _ in range(n - 1)), reverse=True)

cnt = 0
while len(others) > 0 and others[0] >= dasom:
    others[0] -= 1
    dasom += 1
    cnt += 1
    others.sort(reverse=True)

print(cnt)

File: 백준/Silver/1417. 국회의원 선거/국회의원 선거.js
const fs = require('fs'); 
let [n, dasom, ...others] = fs.readFileSync(0, 'utf-8').toString().trim().split('\n').map(Number);

others.sort((a, b) => b - a);

let count = 0;
while (others.length > 0 && others[0] >= dasom) {
    others[0]--;
    dasom++;
    count++;

    others.sort((a, b) => b - a);
}

console.log(count);